### ISSUE TITLE
[BUG] Apex otimized SyncBatchnormFunction crashes with track_running_stats=False

### ISSUE THREAD

**Issue by OlivierDehaene** [NVIDIA/apex#1071]
If `track_running_stats=False` and the layer is in training mode, `SyncBatchnormFunction` crashes as it tries to check the `dtype` of `running_mean` and `running_variance` that are both None.

The bug is located here: [https://github.com/NVIDIA/apex/blob/master/apex/parallel/optimized_sync_batchnorm.py#L85](https://github.com/NVIDIA/apex/blob/master/apex/parallel/optimized_sync_batchnorm.py#L85). 

`self.training` should not be taken into account when passing the `track_runing_stats` argument to the function as the documentation states that "when track_running_stats is  set to ``False``, this module does not track such statistics and always uses batch statistics in both **training** and eval modes."

**Comment by OlivierDehaene** [comment]
I'm available to work on this if you need. The patch should be very fast.
